vlog4j-client/src/main/java/org/vlog4j/client/picocli/VLog4jClientMaterialize.java

```diff
@@ -0,0 +1,187 @@
+
+	@Option(names = "--timeout", description = "Timeout in seconds. Infinite by default", required = false)
+	private int timeout = 0;
+
+	@Option(names = "--query", description = "Positive not-ground Literals to query after materialization in rls syntax. Vlog4jClient will print the size of its extension", required = true)
+	private List<String> queryStrings = new ArrayList<>();
+
+	@ArgGroup(exclusive = false)
+	private PrintQueryResults printQueryResults = new PrintQueryResults();
+
+	@ArgGroup(exclusive = false)
+	private SaveQueryResults saveQueryResults = new SaveQueryResults();
+
+	// TODO save model
+	// @ArgGroup(exclusive = false)
+	// private SaveModel saveModel = new SaveModel();
+
+	@Override
+	public void run() {
+		ExamplesUtils.configureLogging();
+
+		if (!printQueryResults.isConfigOk()) {
+			printQueryResults.printErrorAndExit();
+		}
+
+		if (!saveQueryResults.isConfigOk()) {
+			saveQueryResults.printErrorAndExit();
+		}
+
+		// TODO
+		// if (!saveModel.isConfigOk()) {
+		// saveModel.printErrorAndExit();
+		// }
+
+		System.out.println("Configuration:");
+
+		/* Configure rules */
+		final KnowledgeBase kb = new KnowledgeBase();
+		for (String ruleFile : ruleFiles) {
+			try {
+				RuleParser.parseInto(kb, new FileInputStream(ruleFile));
+				System.out.println("  --rule-file: " + ruleFile);
+			} catch (FileNotFoundException e) {
+				System.out.println("File not found: " + ruleFile + ". " + e.getMessage());
+				System.exit(1);
+			} catch (ParsingException e) {
+				System.out.println("Failed to parse rule file: " + ruleFile + ". " + e.getMessage());
+				System.exit(1);
+			}
+		}
+
+		/* Configure queries (We throw parsing query errors before materialization) */
+		List<PositiveLiteral> queries = new ArrayList<>();
+		for (String queryString : queryStrings) {
+			try {
+				queries.add(RuleParser.parsePositiveLiteral(queryString));
+				System.out.println("  --query: " + queries.get(queries.size() - 1));
+			} catch (ParsingException e) {
+				System.out.println("Failed to parse query: " + queryString + ". " + e.getMessage());
+				System.exit(1);
+			}
+		}
+
+		/* Print configuration */
+		System.out.println("  --log-file: " + logFile);
+		System.out.println("  --log-level: " + logLevel);
+		System.out.println("  --chase-algorithm: " + chaseAlgorithm);
+		System.out.println("  --timeout: " + ((timeout > 0) ? Integer.toString(timeout) : "none"));
+
+		/* Print what to do with the result */
+
+		printQueryResults.printConfiguration();
+		saveQueryResults.printConfiguration();
+		// saveModel.printConfiguration();
+
+		try (Reasoner reasoner = new VLogReasoner(kb)) {
+			// logFile
+			reasoner.setLogFile(logFile);
+			// logLevel
+			reasoner.setLogLevel(logLevel);
+			// chaseAlgorithm
+			reasoner.setAlgorithm(chaseAlgorithm);
+			// timeout
+			if (timeout > 0) {
+				reasoner.setReasoningTimeout(timeout);
+			}
+
+			try {
+				System.out.println("Executing the chase ...");
```

Comment:
remove this line from the try block, the try blocks should contain only what could cause the exception.

Reply:
Thank you. This was a silly mistake